client/task_runner.go

```diff
@@ -34,6 +35,12 @@ const (
 	killFailureLimit = 5
 )
 
+// TaskStatsReporter exposes APIs to query resource usage of a Task
+type TaskStatsReporter interface {
+	ResourceUsage() *cstructs.TaskResourceUsage
```

Comment:
Adding comments, not sure the suggested names are any better than the ones I have there, especially `AllResourceUsage` doesn't imply that instead of returning the latest we are retuning a time series.